torch/testing/_internal/common_utils.py

```diff
@@ -803,204 +810,411 @@ def is_integral(self, dtype):
+                                                               exact_device=exact_device)
+
+                if not values_result and message is None:
+                    message = "Sparse tensor values failed to compare as equal! " + debug_msg
+                self.assertTrue(values_result, msg=message)
             elif x.is_quantized and y.is_quantized:
                 self.assertEqual(x.qscheme(), y.qscheme(), atol=atol, rtol=rtol,
-                                 message=message, allow_inf=allow_inf, exact_dtype=exact_dtype)
+                                 message=message, exact_dtype=exact_dtype,
+                                 exact_device=exact_device)
+
                 if x.qscheme() == torch.per_tensor_affine:
                     self.assertEqual(x.q_scale(), y.q_scale(), atol=atol, rtol=rtol,
-                                     message=message, allow_inf=allow_inf, exact_dtype=exact_dtype)
+                                     message=message, exact_dtype=exact_dtype,
+                                     exact_device=exact_device)
                     self.assertEqual(x.q_zero_point(), y.q_zero_point(),
                                      atol=atol, rtol=rtol, message=message,
-                                     allow_inf=allow_inf, exact_dtype=exact_dtype)
+                                     exact_dtype=exact_dtype, exact_device=exact_device)
                 elif x.qscheme() == torch.per_channel_affine:
                     self.assertEqual(x.q_per_channel_scales(), y.q_per_channel_scales(), atol=atol, rtol=rtol,
-                                     message=message, allow_inf=allow_inf, exact_dtype=exact_dtype)
+                                     message=message, exact_dtype=exact_dtype,
+                                     exact_device=exact_device)
                     self.assertEqual(x.q_per_channel_zero_points(), y.q_per_channel_zero_points(),
                                      atol=atol, rtol=rtol, message=message,
-                                     allow_inf=allow_inf, exact_dtype=exact_dtype)
+                                     exact_dtype=exact_dtype, exact_device=exact_device)
                     self.assertEqual(x.q_per_channel_axis(), y.q_per_channel_axis(),
-                                     atol=atol, rtol=rtol, message=message)
-                self.assertEqual(x.dtype, y.dtype)
-                self.assertEqual(x.int_repr().to(torch.int32),
-                                 y.int_repr().to(torch.int32), atol=atol, rtol=rtol,
-                                 message=message, allow_inf=allow_inf, exact_dtype=exact_dtype)
+                                     atol=atol, rtol=rtol, message=message,
+                                     exact_dtype=exact_dtype, exact_device=exact_device)
+
+                # TODO: the following line should be removed once exact_dtype
+                # defaults to True.
+                self.assertTrue(x.dtype is y.dtype, msg=message)
```

Comment:
should this be wrapped with an `if exact_dtype`? Why does changing the default imply removing the check?

Reply:
The check was pre-existing. So it should be wrapped like you're suggesting, but because exact_dtype is often false it would regress test coverage for quantized tensors. That's why I thought we'd put the check in (which is correct) once the default was true.

Essentially quantized tensors were already demanding exact_dtype = True.